docs/specifications/tests/Zone-TP/zone09.md

```diff
@@ -1,47 +1,278 @@
+
+* "Child Zone" - The domain name to be tested.
+
+
+## Summary
+
+* Basic name server problems are not reported by this test case.
+* A notify is issued if MX is missing, except for root, an ARPA zone and a TLD.
+* A warning is issued if a TLD *has* an MX.
+
+Message Tag outputted               | Level   | Arguments               | Description of when message tag is outputted
+:-----------------------------------|:--------|:------------------------|:--------------------------------------------
+Z09_INCONSISTENT_MX_RRSETS          | WARNING |                         | Some name servers returns MX RRset and other none.
+Z09_MISSING_MAIL_TARGET             | NOTICE  |                         | The child zone has no mail target (no MX).
+Z09_MX_DATA                         | INFO    | ns_ip_list, domain_list | The mail targets in the MX RRset returned by some name servers, and a list of those.
+Z09_MX_FOUND                        | INFO    | ns_ip_list              | List of name servers that return MX RRset.
+Z09_NON_AUTHORITATIVE_MX_RESPONSE   | WARNING | ns_ip_list              | List of name servers that gave non-authoritative response on the MX query.
+Z09_NO_MX_FOUND                     | INFO    | ns_ip_list              | List of name servers that did not return MX RRset.
+Z09_NO_RESPONSE_MX_QUERY            | WARNING | ns_ip_list              | List of name servers that did not respond on MX query.
+Z09_NULL_MX_WITH_NON_ZERO_PREFERENCE| NOTICE  |                         | The child zone has a null MX with incorrect preference (must be 0).
+Z09_NULL_MX_WITH_OTHER_MX           | WARNING |                         | The child zone has a null MX but incorrectly mixed with other MX records.
+Z09_ROOT_EMAIL_DOMAIN_MEANINGLESS   | NOTIFY  |                         | The root zone has an MX RRset, which is meaningless.
+Z09_TLD_EMAIL_DOMAIN_HARMFUL        | WARNING |                         | The child zone is a TLD and has MX, which is considered harmful.
+Z09_UNEXPECTED_RCODE_MX_RESPONSE    | WARNING | ns_ip_list, rcode       | The name servers listed returns unexpected RCODE value (listed) on the MX query.
+
+The value in the Level column is the default severity level of the message. The
+severity level can be changed in the [Zonemaster-Engine profile]. Also see the
+[Severity Level Definitions] document.
+
+The argument names in the Arguments column lists the arguments used in the
+message. The argument names are defined in the [argument list].
+
+
+## Test procedure
+
+1.  Create an SOA query for the *Child Zone* without OPT record (no EDNS)
+    ("SOA Query").
+
+2.  Create an MX query for the *Child Zone* without OPT record (no EDNS)
+    ("MX Query").
+
+3.  Obtain the set of name server IP addresses using [Method4] and [Method5]
+    ("Name Server IP").
+
+4. Create the following empty sets
+
+    1.  Name server IP addresses ("No Response MX Query").
+    2.  Name server IP addresses and RCODE value ("Unexpected RCODE MX Query").
```

Comment:
The data type of the set isn't clear from the description. Maybe something like this?
```suggestion
    2.  Name server IP address and RCODE value pairs ("Unexpected RCODE MX Query").
```